src/lib/operators/table_scan/column_is_null_table_scan_impl.cpp

```diff
@@ -29,43 +30,29 @@ std::shared_ptr<PosList> ColumnIsNullTableScanImpl::scan_chunk(const ChunkID chu
 
   auto matches = std::make_shared<PosList>();
 
-  _scan_non_reference_segment(*segment, chunk_id, *matches, nullptr);
+  if (const auto value_segment = std::dynamic_pointer_cast<BaseValueSegment>(segment)) {
+    _scan_value_segment(*value_segment, chunk_id, *matches, nullptr);
+  } else {
+    _scan_non_value_segment(*segment, chunk_id, *matches, nullptr);
+  }
 
   return matches;
 }
 
-void ColumnIsNullTableScanImpl::_scan_non_reference_segment(
-    const BaseSegment& segment, const ChunkID chunk_id, PosList& matches,
-    const std::shared_ptr<const PosList>& position_filter) const {
-  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
-    _scan_segment(typed_segment, chunk_id, matches, position_filter);
-  });
-}
-
-void ColumnIsNullTableScanImpl::_scan_segment(const BaseSegment& segment, const ChunkID chunk_id, PosList& matches,
-                                              const std::shared_ptr<const PosList>& position_filter) const {
-  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
-    using Type = typename decltype(type)::type;
-    auto iterable = create_iterable_from_segment<Type>(typed_segment);
-
+void ColumnIsNullTableScanImpl::_scan_non_value_segment(const BaseSegment& segment, const ChunkID chunk_id,
+                                                        PosList& matches,
+                                                        const std::shared_ptr<const PosList>& position_filter) const {
+  segment_with_iterators(segment, position_filter, [&](auto it, const auto end) {
```

Comment:
I had to read this a couple of times before I saw that `segment_with_iterators` also takes a position filter. Maybe we should include that in the name?

Reply:
Even though this made me spell out the hideous word "position_filter", I renamed it to "segment_with_iterators_and_position_filter" ;)

Reply:
I hate names. Maybe we should compile everything, decompile it, and live with random names.

`segment_with_iterators_filtered`?

Reply:
Maybe that's a tiny bit better...